### PR TITLE
Backend work

### PR DESCRIPTION
**Pull request by ahmedqarmout2**
This includes some changes to the routes, nothing of which should crash any existing routes except the deletion of users and questions(need to decide on that later). I am still working on backend changes, but this will unblock everyone else

### CODE REVIEW

**Comment by esteeroramy** [comment]
Let's also update the documentation files so that it matches these changes

File: public/js/login.js [hunk 1]
```diff
@@ -2,14 +2,14 @@
 $('#login').submit(function(evt) {
     evt.preventDefault();
     $.ajax({
-        type: 'POST',
+        type: 'GET',
```
Comment (Mohammed-JS-Khan) [inline public/js/login.js:5]: Logins requests are generally supposed to be POST requests so the username and password go through the body and not the URL. Is there a specific reason why we changed it to a GET?

Comment (ahmedqarmout2) [reply public/js/login.js:5]: https://stackoverflow.com/questions/630453/put-vs-post-in-rest

Put and post are used to update and create resources on the backend. When you login, you don't add to the backend, you just fetch

File: server/questions.js [hunk 2]
```diff
@@ -18,149 +18,174 @@ You should have received a copy of the GNU General Public License
...
+	questionToAdd.points = question.points;
+	questionToAdd.type = common.questionTypes.MULTIPLECHOICE;
+	questionToAdd.attempted = [];
+	questionToAdd.answered = [];
+	questionToAdd.attempts = [];
+	questionToAdd.ctime = currentDate;
+	questionToAdd.mtime = currentDate;
+	db.addMultipleChoiceQuestion(questionToAdd, function(err, res) {
+		callback(err, res);
+	});
```
Comment (mehtadarshan1) [inline server/questions.js:83]: Can you add questionToAdd.answerOptions --> dictionary to store different answer options

File: server/questions.js [hunk 3]
```diff
@@ -18,149 +18,174 @@ You should have received a copy of the GNU General Public License
...
+				userId,
+				{ questionId:questionId, correct:value, points:question.points },
+				function(err, res){
+					updateQuestionByIdWithRedirection(
+						questionId,
+						{ userId:userId, correct:value },
+						function(err, res) {
+							callback(err, value);
+						}
+					);
```
Comment (mehtadarshan1) [inline server/questions.js:186]: can you fix the indentation here

**Review by mehtadarshan1 — CHANGES_REQUESTED** [review]

File: public/js/login.js (hunk 1 above)
Comment (Mohammed-JS-Khan) [reply public/js/login.js:5]: Technically you're "creating" a session in the backend when you're logging in (even though we don't have sessions). POST give a more secure way of executing a login and it not only keeps the URL clean looking, it prevents your salted and hashed password from being starkly visible in the URL, prevents against browser prefetching, and prevents against search engine spidering. See the following:

https://stackoverflow.com/questions/15098392/which-http-method-should-login-and-logout-actions-use-in-a-restful-setup

https://stackoverflow.com/questions/3521290/logout-get-or-post/14587231#14587231

https://stackoverflow.com/questions/7140074/restfully-design-login-or-register-resources

https://stackoverflow.com/questions/5868786/what-method-should-i-use-for-a-login-authentication-request

https://stackoverflow.com/questions/2001773/understanding-rest-verbs-error-codes-and-authentication

Comment (ahmedqarmout2) [reply public/js/login.js:5]: As we discussed offline, it's best to use post. Will fix

File: server/questions.js (hunk 2 above)
Comment (ahmedqarmout2) [reply server/questions.js:83]: Will do

**Pull request merged**